### PR TITLE
9395: deliveries v1 api

### PR DESCRIPTION
**Pull request by mkalish**
This PR adds the V1 API for fetching delivered reports to receivers

Test Steps:
1. Make sure that your DB has some reports recorded that were sent to a receiver
2. Use the following to test the API
```sh
curl --location 'http://localhost:7071/api/v1/receivers/ignore.HL7/deliveries' \
--header 'Content-Type: application/json' \
--data '{
    "sort": {
        "direction": "ASC",
        "property": "number_of_tests"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": [
        {
            "value": "2023-05-21T00:00:00+00:00",
            "filterName": "UNTIL"
        }
    ]
}'
```
Refer to the `delivery.yml` file for the parameters that you can use with the API
3. Verify that the returned report IDs were delivered to the receiver
4. Verify that the data on provider, facility, sender and test numbers are all correct for the delivered reports (I tested this by sending data checking the results, resetting the db and sending data with different provider, facility, etc.)

## Changes
- Adds a V1 API for fetching reports delivered to receivers

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #9395

### CODE REVIEW

File: prime-router/src/main/kotlin/history/db/ReportGraph.kt
Comment (mkalish) [inline prime-router/src/main/kotlin/history/db/ReportGraph.kt:148]: Need to keep track of the starting report so that I can return it as part of the delivery API

**Comment by mkalish** [comment]
The conflict will get resolved after the stacked PR is merged

**Review by jack-h-wang — APPROVED** [review]
Looks good to me! Was able to confirm returned results and pagination works as expected.

**Review by arnejduranovic — APPROVED** [review]
Looks good minus one small comment, but let's wait for approval from Experience team before merging

File: prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt
Comment (mkalish) [reply prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt:127]: Yea... I'm had been thinking the same thing, but needed to put some thought into if there was an easy way to generalize returning the error message. 

I'll try to include that in this PR

**Comment by etanb** [comment]
@mkalish is it possible to include an `all` option or some equivalent in the Pagination _just in case_ we want to get all the results at once?

**Comment by mkalish** [comment]
@etanb I think from a performance standpoint that would be tough.  Curious what situation you be covering with that?

**Comment by etanb** [comment]
@mkalish From my understanding, a lot of times the number of results aren't significant enough to slow down rendering on the client-side, so just wanted to give us options in how we rendered the data in the future. Is it a big lift to add this? Would doing something like `limit: 99999999` effectively do this?

**Comment by mkalish** [comment]
@etanb Yea, my recommendation would be to adjust the limit to the amount of the data the front end wants to display which can be arbitrarily high.

Overall though, I think we would want to have a conversation around this and use case since there are also performance implications for loading that much data in the backend.

**Review by etanb — APPROVED** [review]

**Comment by etanb** [comment]
@mkalish is it possible to tell what the biggest possible dataset we have that would be called? the FE could handle many rows of data, so just wanted to see if this is a problem that doesn't quite exist yet. Would love to eventually have an `all` option but LGTM.

File: prime-router/docs/api/delivery.yml
Comment (snesm) [inline prime-router/docs/api/delivery.yml:11]: POST parameters do not occur in the path.

This is the first API to use a versioning based path.  Is there an agreement that we will use this format moving forward?  Is there a plan on how to handle the existing API's (add a /v1/ alias)?

File: prime-router/docs/api/delivery.yml
Comment (snesm) [inline prime-router/docs/api/delivery.yml:16]: securitySchemes is missing

```
  securitySchemes:
    OAuth2:
      $ref: 'https://raw.githubusercontent.com/CDCgov/prime-reportstream/master/prime-router/docs/api/components/security_schemes.yml#/OAuth2'
```

File: prime-router/docs/api/delivery.yml
Comment (snesm) [inline prime-router/docs/api/delivery.yml:44]: POST parameters do not occur in the path.

**Review by snesm — CHANGES_REQUESTED** [review]

**Comment by luis-pabon-tf** [comment]
Screwed up the diff. Gonna fix that up real quick.....

**Comment by luis-pabon-tf** [comment]
Didn't mean to merge it before responding to @snesm comments. Will respond to them now!

File: prime-router/docs/api/delivery.yml
Comment (luis-pabon-tf) [reply prime-router/docs/api/delivery.yml:16]: We've been using `security` and `securitySchemes` somewhat interchangeably on our API docs. I think it's worth discussing the structure of those components. 
I noticed there's different use cases for each: https://swagger.io/docs/specification/authentication/

**Pull request merged**
